src/main/scala/tech/cryptonomic/conseil/routes/PlatformDiscovery.scala

```diff
@@ -58,26 +58,27 @@ class PlatformDiscovery(config: PlatformsConfiguration, tezosPlatformDiscoveryOp
       }
   }
 
+  import cats.instances.future._
+  import cats.instances.option._
+  import cats.syntax.traverse._
   /** Metadata route implementation for attributes values endpoint */
-  private val attributesValuesRoute = attributesValuesEndpoint.implementedByAsync {
+  private lazy val attributesValuesRoute = attributesValuesEndpoint.implementedByAsync {
     case ((platform, network, entity), attribute, _) =>
-      tezosPlatformDiscoveryOperations.listAttributeValues(entity, attribute).map { attributes =>
-        ConfigUtil.getNetworks(config, platform).find(_.network == network).map { _ =>
+      ConfigUtil.getNetworks(config, platform).find(_.network == network).map { _ =>
+        tezosPlatformDiscoveryOperations.listAttributeValues(entity, attribute).map { attributes =>
           attributes
         }
-      }
+      }.sequence
```

Comment:
This can thus become a single `traverse`?

```scala
...getNetworks(...).find(...).traverse { _ =>
      tezosPlatformDiscorveryOperations.listAttributeValues(...)
}
```

Reply:
changed